src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs

```diff
@@ -133,22 +146,75 @@ Size IAnimatedVisualSource.Measure(Size availableSize)
 			var availableWidth = availableSize.Width;
 			var availableHeight = availableSize.Height;
 
+			var resultSize = availableSize;
+
 			if (double.IsInfinity(availableWidth))
 			{
 				if (double.IsInfinity(availableHeight))
 				{
 					return compositionSize;
 				}
 
-				return new Size(availableHeight * compositionSize.Width / compositionSize.Height, availableHeight);
+				resultSize = new Size(availableHeight * compositionSize.Width / compositionSize.Height, availableHeight);
 			}
 
 			if (double.IsInfinity(availableHeight))
 			{
-				return new Size(availableWidth, availableWidth * compositionSize.Height / compositionSize.Width);
+				resultSize = new Size(availableWidth, availableWidth * compositionSize.Height / compositionSize.Width);
+			}
+
+			InnerMeasure(resultSize);
+
+			return resultSize;
+		}
+
+		partial void InnerMeasure(Size size);
+
+		private bool TryLoadEmbeddedJson(Uri uri, out string json)
+		{
+			if (uri.Scheme != "embedded")
+			{
+				json = null;
+				return false;
+			}
+
+			var assemblyName = uri.Host;
+
+			Assembly assembly;
+			if (assemblyName == ".")
+			{
+				assembly = Application.Current.GetType().Assembly;
+			}
+			else
+			{
+				assembly = AppDomain
+					.CurrentDomain.GetAssemblies()
+					.FirstOrDefault((Assembly a) =>
+						a.GetName().Name.Equals(assemblyName, StringComparison.OrdinalIgnoreCase));
```

Comment:
The use of `Assembly.Load(assemblyName)` may be faster than scanning all domain assemblies, as runtimes do that completely natively.

Reply:
I'll try that.

Reply:
Done